### ISSUE TITLE
Failed: Could not create directory,  file exists and is not a directory: / Multi-jvm / assembly / assembledMappings 

### ISSUE THREAD

**Issue by johanandren** [akka/akka#27661]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/7278/console

```
[error] java.lang.RuntimeException: Could not create directory /localhome/jenkinsakka/workspace/akka-multi-node-nightly/akka-cluster-typed/target/streams/multi-jvm/assemblyOption/$global/streams/assembly/7c9e372c6314ee0628ee1dc84f3ff20a457e8168_af7e2274bc4c9d6aab59e1b84f68383d14ab7935_da39a3ee5e6b4b0d3255bfef95601890afd80709
[error] 	at scala.sys.package$.error(package.scala:26)
[error] 	at sbt.io.IO$.createDirectory(IO.scala:326)
[error] 	at sbtassembly.AssemblyUtils$.unzipStream(AssemblyUtils.scala:42)
[error] 	at sbtassembly.AssemblyUtils$.$anonfun$unzip$1(AssemblyUtils.scala:37)
[error] 	at sbt.io.Using.apply(Using.scala:22)
[error] 	at sbtassembly.AssemblyUtils$.unzip(AssemblyUtils.scala:37)
[error] 	at sbtassembly.Assembly$.$anonfun$assembleMappings$9(Assembly.scala:220)
[error] 	at scala.collection.parallel.AugmentedIterableIterator.map2combiner(RemainsIterator.scala:112)
[error] 	at scala.collection.parallel.AugmentedIterableIterator.map2combiner$(RemainsIterator.scala:109)
[error] 	at scala.collection.parallel.immutable.ParVector$ParVectorIterator.map2combiner(ParVector.scala:62)
[error] 	at scala.collection.parallel.ParIterableLike$Map.leaf(ParIterableLike.scala:1052)
[error] 	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:49)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
[error] 	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:63)
[error] 	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:52)
[error] 	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:46)
[error] 	at scala.collection.parallel.ParIterableLike$Map.tryLeaf(ParIterableLike.scala:1049)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal(Tasks.scala:156)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal$(Tasks.scala:153)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.internal(Tasks.scala:436)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:146)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:145)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:436)
[error] 	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
[error] 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[error] 	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
[error] 	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
[error] 	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
[error] (akka-cluster-typed / Multi-jvm / assembly / assembledMappings) Could not create directory /localhome/jenkinsakka/workspace/akka-multi-node-nightly/akka-cluster-typed/target/streams/multi-jvm/assemblyOption/$global/streams/assembly/7c9e372c6314ee0628ee1dc84f3ff20a457e8168_af7e2274bc4c9d6aab59e1b84f68383d14ab7935_da39a3ee5e6b4b0d3255bfef95601890afd80709
[error] Total time: 3496 s, completed Sep 7, 2019 6:26:29 AM
```

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/3583/console

**Comment by patriknw** [comment]
Might be duplicate of https://github.com/akka/akka/issues/27548

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/3584/console

**Comment by helena** [comment]
yes this does not seem to be related to typed or assembly, I see it in many other modules

**Comment by helena** [comment]
Ok this is more like:

[error] (akka-remote-tests / Multi-jvm / assembly / assembledMappings) Could not create directory /localhome/jenkinsakka/workspace/akka-multi-node/akka-remote-tests/target/streams/multi-jvm/assemblyOption/$global/streams/assembly/6ad988c3500ee07fb6a17fa8e90d442c8a2646ba_8cc85d866f3650dfe73b2a0320d8a0f8c9add975_da39a3ee5e6b4b0d3255bfef95601890afd80709: file exists and is not a directory.

**Comment by helena** [comment]
I am starting to think these errors have something to do with the photo 3 upgrade
* the autogenerated file name
     - 6ad988c3500ee07fb6a17fa8e90d442c8a2646ba_8cc85d866f3650dfe73b2a0320d8a0f8c9add975_da39a3ee5e6b4b0d3255bfef95601890afd8070
* recent dependency change
* shading and assembly tasks are always sporty

https://jenkins.akka.io:8498/job/akka-multi-node/3584/consoleFull 
1. 2 multi-node issues
2. Same sbt task:   / Multi-jvm / assembly / assembledMappings) 
2. Same path, but different module: all the way down to the autogenerated file/dir name:
     - Could not create directory /localhome/jenkinsakka/workspace/akka-multi-node/${module_name}/target/streams/multi-jvm/assemblyOption/$global/streams/assembly/6ad988c3500ee07fb6a17fa8e90d442c8a2646ba_8cc85d866f3650dfe73b2a0320d8a0f8c9add975_da39a3ee5e6b4b0d3255bfef95601890afd80709

1 from cluster-tools is just a warning, but referencing protobuf
* [warn] Could not create directory /localhome/jenkinsakka/workspace/akka-multi-node/akka-cluster-tools/target/streams/multi-jvm/assemblyOption/$global/streams/assembly/6ad988c3500ee07fb6a17fa8e90d442c8a2646ba_8cc85d866f3650dfe73b2a0320d8a0f8c9add975_da39a3ee5e6b4b0d3255bfef95601890afd80709/google/protobuf

1 error (reported twice for same issue) remote-tests
*  [error] (akka-remote-tests / Multi-jvm / assembly / assembledMappings) Could not create directory /localhome/jenkinsakka/workspace/akka-multi-node/akka-remote-tests/target/streams/multi-jvm/assemblyOption/$global/streams/assembly/6ad988c3500ee07fb6a17fa8e90d442c8a2646ba_8cc85d866f3650dfe73b2a0320d8a0f8c9add975_da39a3ee5e6b4b0d3255bfef95601890afd80709: file exists and is not a directory.
* [error] java.lang.RuntimeException: Could not create directory /localhome/jenkinsakka/workspace/akka-multi-node/akka-remote-tests/target/streams/multi-jvm/assemblyOption/$global/streams/assembly/6ad988c3500ee07fb6a17fa8e90d442c8a2646ba_8cc85d866f3650dfe73b2a0320d8a0f8c9add975_da39a3ee5e6b4b0d3255bfef95601890afd80709: file exists and is not a directory.

**Comment by helena** [comment]
Can kind of see the issue here 
https://jenkins.akka.io:8498/job/akka-multi-node-repeat-helena/ws/akka-remote-tests/target/streams/multi-jvm/assemblyOption/$global/streams/assembly/

**Comment by eed3si9n** [comment]
@helena https://github.com/sbt/sbt-assembly/blob/v0.14.9/src/main/scala/sbtassembly/Assembly.scala#L201-L228 shows that sbt-assembly tries to unzip bunch of things in parallel. I wonder if this would blow up when there are duplicates in the classpath.

**Comment by helena** [comment]
Thanks @eed3si9n, and we do have duplicate names applied, e.g. here a `directory` and a `.jarName`

00c5f3c14fb073be5a5b2a003f7e0f503e94c7c6_1b035a1a4ce5d3441a4a1a331d04839ef487ec49_da39a3ee5e6b4b0d3255bfef95601890afd80709

**Comment by patriknw** [comment]
also note multiple entries of the 

```
[info] Including: akka-protobuf-v3-assembly-2.6-SNAPSHOT.jar
[info] Including: junit-4.12.jar
[info] Including: akka-protobuf-v3-assembly-2.6-SNAPSHOT.jar
[info] Including: akka-protobuf-v3-assembly-2.6-SNAPSHOT.jar
[info] Including: akka-protobuf-v3-assembly-2.6-SNAPSHOT.jar
[info] Including: akka-protobuf-v3-assembly-2.6-SNAPSHOT.jar
[info] Including: akka-protobuf-v3-assembly-2.6-SNAPSHOT.jar
```

followed by
```
[warn] Could not create directory /localhome/jenkinsakka/workspace/akka-multi-node/akka-cluster-tools/target/streams/multi-jvm/assemblyOption/$global/streams/assembly/6ad988c3500ee07fb6a17fa8e90d442c8a2646ba_8cc85d866f3650dfe73b2a0320d8a0f8c9add975_da39a3ee5e6b4b0d3255bfef95601890afd80709/google/protobuf
```

**Comment by helena** [comment]
Yes, and I was wondering if we can make it build only once vs each time

**Comment by patriknw** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/3605

**Comment by patriknw** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/7302

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/3626/consoleFull

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-artery-aeron-udp-remote-tests/142/console

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/7321/console
https://jenkins.akka.io:8498/job/akka-artery-cluster-tests/2124/console

**Comment by jrudolph** [comment]
Do we have any idea how those multiple `Including: akka-protobuf-v3-assembly-2.6-SNAPSHOT.jar` would end up on the classpath assembly is trying to collect together?

Has someone managed to reproduce that part?

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/3636/console

**Comment by patriknw** [comment]
could the reason be as simple as we have multiple dependsOn protobufV3 from various places? If we can have only one? I think `stream` can be the root that all others anyway depends on. Maybe `persistence` is not depending on `stream` but wouldn't hurt too much to add that dependency anyway?

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/7330/consoleText

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/3648/console

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/7342/consoleText (hopefully this was the last one then with the potential fix merged)

**Comment by helena** [comment]
Thanks @patriknw, we were hoping :) I planned to fix it this week, but was sick. I'll take it from here.

**Comment by patriknw** [comment]
Maybe we should update the sbt-assembly version in sbt-multi-jvm?

Given the `parallel` in stack trace it might be a concurrency bug?

**Comment by chbatey** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/7354/consoleFull

**Comment by patriknw** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/7366
